uvtools/dspec.py

```diff
@@ -315,73 +354,114 @@ def fourier_filter(x, data, wgts, filter_centers, filter_half_widths, suppressio
                    if ndim_data == 1:
                        data = np.asarray([data])
                        wgts = np.asarray([wgts])
-                   if not filter2d and filter_dim == 0:
-                       data = data.T
-                       wgts = wgts.T
+                   if len(filter_dims) == 2:
+                       filter2d = True
+                   elif len(filter_dims) == 1:
+                       filter2d = False
+                   else:
+                     raise ValueError("filter_dims either have length 1 or length 2")
+                   if 0 in filter_dims and not filter2d:
+                     data = data.T
+                     wgts = wgts.T
+                   if 'cache' not in kwargs:
+                      cache = {}
+                   else:
+                      cache = kwargs.pop('cache')
+                   # process kwargs
+                   if 'dayenu' == mode[0]:
+                       if len(mode) > 1:
+                           if 'dft' in mode:
+                               defaults = {**DAYENU_DEFAULTS, **DFT_DEFAULTS}
+                               dicts_2d = DEFAULT_DAYENU_FILT2D + DEFAULT_DFT_FILT2D
+                           elif 'dpss' in mode:
+                               defaults = {**DAYENU_DEFAULTS, **DPSS_DEFAULTS}
+                               dicts_2d = DEFAULT_DAYENU_FILT2D + DEFAULT_DPSS_FILT2D
+
+                       else:
+                           defaults = DAYENU_DEFAULTS
+                           dicts_2d = DEFAULT_DAYENU_FILT2D
+                   elif mode[0] == 'dft':
+                        defaults = DFT_DEFAULTS
+                        dicts_2d = DEFAULT_DFT_FILT2D
+                   elif mode[0] == 'dpss':
+                        defaults = DPSS_DEFAULTS
+                        dicts_2d = DEFAULT_DPSS_FILT2D
+                   elif mode[0] == 'clean':
+                        defaults = CLEAN_DEFAULTS
+                        dicts_2d = DEFAULT_CLEAN_FILT2D
+                   _process_filter_kwargs(kwargs, defaults, filter2d, dicts_2d)
+                   if 'dft' in mode:
+                        fp = np.asarray(kwargs['fundamental_period']).flatten()
+                        for m in range(len(fp)):
+                            if np.isnan(fp[m]):
+                                fp[m] = 2. * (x[m].max() - x[m].min())
+                        if len(fp) == 1:
+                            kwargs['fundamental_period'] = fp[0]
+                        else:
+                            kwargs['fundamental_period'] = list(fp)
+
                    if mode[0] == 'dayenu':
                        if filter2d:
-                           filter_dim_d = [1, 0]
+                          filter_dims_d = [1, 0]
                        else:
-                           filter_dim_d = [1]
-                       residual, info = dayenu_filter(x=x, data=data, wgts=wgts, filter_dimensions=filter_dim_d,
+                          filter_dims_d = [1]
```

Comment:
Did you address this?

Reply:
filter_dims = 0 means that we should also filter across the 0 (time) dimension of the array.

Reply:
In addition, if the flags are not separable, then the order of the dimensions to filter do not commute. In this case, it is better to clean the frequency dimension first because the dynamic range between foregrounds and signal tends to be larger then the dynamic range between cross-talk and signal.

Reply:
If your comment was "what if..." then, the answer is that on line 384, the data arrays are transposed so that all 1d filters are carried out as if we are filtering the frequency dimension.

Reply:
Sorry, yeah, I meant "what if". That clears things up. Perhaps add a comment here explaining.

Reply:
Cool. I do think this is a bit unclear so I've added comments here (and in other places where dimensions are set manuall to [1] and the transposes) to make sure readers down the road don't get confused.